File: backend/app/schemas/library.py
from pydantic import BaseModel

from app.schemas.base import TimestampedModel


class LibraryBase(BaseModel):
    name: str
    path: str
    enabled: bool = True


class LibraryCreate(LibraryBase):
    pass


class LibraryUpdate(BaseModel):
    name: str | None = None
    path: str | None = None
    enabled: bool | None = None


class LibraryRead(LibraryBase, TimestampedModel):
    id: str

    model_config = {
        "from_attributes": True,
    }

File: frontend/lib/types/library.ts
export interface LibraryRead {
  id: string;
  name: string;
  path: string;
  enabled: boolean;
  created_at?: string | null;
  updated_at?: string | null;
}

export interface LibraryCreate {
  name: string;
  path: string;
  enabled?: boolean;
}

export interface LibraryUpdate {
  name?: string | null;
  path?: string | null;
  enabled?: boolean | null;
}
